frontend/src/actions/users.ts

```diff
@@ -285,6 +299,40 @@ export const userUpdateProfileAvatar = (data: {
   };
 };
 
+export const fetchLearnedWords = (data: {
+  token: string;
+  callback?: Function;
+  errorCallback?: Function;
+  finallyCallback?: Function;
+}) => {
+  return async (dispatch: Dispatch) => {
+    backend.get('/sanctum/csrf-cookie').then(async (csrf_response) => {
+      let { token, callback, errorCallback, finallyCallback } = data;
+
+      await backend
+        .get<LearnedWordsData>(`/api/learned_words`, {
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        })
+        .then((response: AxiosResponse) => {
+          if (callback) callback(response);
+
+          dispatch<FetchLearnedWordsAction>({
+            type: ActionTypes.fetchLearnedWords,
+            payload: response.data,
+          });
+        })
+        .catch((err: AxiosError) => {
+          if (errorCallback) errorCallback(err);
+        })
+        .finally(() => {
+          if (finallyCallback) finallyCallback();
+        });
```

Comment:
This is working normal. Base on the previous projects we use the `axios.create` it creates an instance of axios and accepts a config as a parameter it includes the method,url, and payload.

No need to change this since you are using this on all your files.

Reply:
baseAPI.js
```js
import axios from 'axios';
import Cookies from 'js-cookie';

const TOKEN = Cookies.get('access_token');

const API = axios.create({
  baseURL: `${process.env.REACT_APP_SERVER_API}/api`,
  headers: {
    'Content-Type': 'application/json',
    'Authorization': `Bearer ${TOKEN}`,
  }
});

export default API;
```
savequizAPI.js
```js
import API from './base';

const SaveQuizAPI = {
  save: (payload) => {
    const options = {
      method: 'POST',
      url: '/save',
      data: payload,
    };

    return API.request(options);
  },
};

export default SaveQuizAPI;
```

Reply:
yes sir, i have noted this already, however, i do have some calls that were made earlier prior to knowing this sir, i will keep this in mind sir thank you